### PR TITLE
fix(useAsyncValidator): fix Schema importing from async-validator

### PR DESCRIPTION
**Pull request by theruziev**
### Description

that PR fixed issue #2497 

### Additional context

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Review by okxiaoliang4 — APPROVED** [review]

File: packages/integrations/useAsyncValidator/index.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ import type { Rules, ValidateError, ValidateOption } from 'async-validator'
 import type { Ref } from 'vue-demi'
 import { computed, ref, watchEffect } from 'vue-demi'
 
+// @ts-expect-error Schema.default is exist in ssr mode
+const AsyncValidatorSchema = Schema || Schema.default
```
[the remark this answers is not in the document]
Comment (theruziev) [reply packages/integrations/useAsyncValidator/index.ts:9]: I found my playground by this PR https://stackblitz.com/edit/github-61yzgr-wtmb2h?file=composable%2Fvalidator%2Findex.ts you can check Schema in SSR mode is not a constructor

[the remark this answers is not in the document]
Comment (sfreytag) [reply packages/integrations/useAsyncValidator/index.ts:9]: I was testing in Node with a Vite-based SSR. If the constructor we want is Schema.default where `typeof Schema.default === 'function` then Schema will be an object, ie `typeof Schema === 'object'`. Therefore the logic `Schema || Schema.default` will return `Schema` which is not what you want. I got it to work by reversing the logic `Schema.default || Schema`. In the browser Schema.default is undefined so I get Schema, which is correct. In Node they're both defined so I get Schema.default which is also correct.

Demo stackblitz here: https://stackblitz.com/edit/node-bdkhusjx

EIther way though, thanks for this fix!

[the remark this answers is not in the document]
Comment (theruziev) [reply packages/integrations/useAsyncValidator/index.ts:9]: Got it. I have another PR #2788 where I was doing what you wrote

[the remark this answers is not in the document]
Comment (sfreytag) [reply packages/integrations/useAsyncValidator/index.ts:9]: Ah perfect, yes. Sorry I did not find that - I guess I did not work up to the head to look at the latest. I am not actually using useAyncValidator, just ran into the same issue with another use of async-validator with SSR, so this gave me the info to solve it.

**Pull request merged**